runtime/tests/runtime_integration_tests/common.rs

```diff
@@ -172,7 +184,12 @@ pub fn execute_dpe_cmd(model: &mut DefaultHwModel, dpe_cmd: &mut Command) -> Res
         &resp[core::mem::size_of_val(&resp_hdr.hdr.chksum)..],
     ));
 
-    parse_dpe_response(dpe_cmd, &resp_hdr.data[..resp_hdr.data_size as usize])
+    let resp_bytes = &resp_hdr.data[..resp_hdr.data_size as usize];
+    Some(match expected_result {
+        DpeResult::Success => parse_dpe_response(dpe_cmd, resp_bytes),
+        DpeResult::DpeCmdFailure => Response::Error(ResponseHdr::read_from(resp_bytes).unwrap()),
+        DpeResult::MboxCmdFailure(_) => panic!("Not possible"),
```

Comment:
nit: You can use the `unreachable!` macro and then have a short message explaining why.

Reply:
Done